app/models/hostgroup.rb

```diff
@@ -212,6 +213,13 @@ def update_ancestry_puppetclasses
     unscoped_find(ancestry.to_s.split('/').last.to_i).update_puppetclasses_total_hosts if ancestry.present?
   end
 
+  def children_hosts_count
+    count = hosts_count
+    return count unless children.any?
+    children.each {|child| count += child.children_hosts_count }
```

Comment:
You can use `children.map(&:children_hosts_count).reduce(hosts_count, :+)` which will work whether or not there are any children.